.github/workflows/ci-vulnerability-scans.yml

```diff
@@ -0,0 +1,103 @@
+# GitHub Actions CI workflow that runs vulnerability scans on the application's Docker image
+# to ensure images built are secure before they are deployed. 
+
+# NOTE: The workflow isn't able to pass the docker image between jobs, so each builds the image.
+#       A future PR will pass the image between the scans to reduce overhead and increase speed
+name: API Linting & Testing CI
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - app/**
+  pull_request:
+    paths:
+        - app/**
+
+jobs:
+  hadolint-scan:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # Scans Dockerfile for any bad practices or issues
+      - name: Scan Dockerfile by hadolint
+        uses: hadolint/hadolint-action@v3.1.0
+        with:
+          dockerfile: app/Dockerfile
+          format: json
+          failure-threshold: warning
+
+  trivy-scan:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build and tag Docker image for scanning
+        id: build-image
+        run: |
+          make release-build
+          echo "image=template-infra-app:latest" >> $GITHUB_OUTPUT
```

Comment:
the image tag will only be `template-infra-app` in this template repo, but in a project repo it'll be `$PROJECT_NAME-$APP_NAME` where `PROJECT_NAME` defaults to the name of the repo directory, and `APP_NAME` defaults to `app` (you can see in the `Makefile` that `release-build` uses those environment variable `IMAGE_NAME` which is set to `$PROJECT_NAME-$APP_NAME`). So we might need to a way to get the actual `IMAGE_NAME` used from the `Makefile`. One idea is to add a `release-image-name` target to the `Makefile` like:

```make
release-image-name:
	@echo $(IMAGE_NAME)
```

then here we do something like
```suggestion
          make release-build
          IMAGE_NAME=$(make release-image-name)
          echo "image=$IMAGE_NAME:latest" >> $GITHUB_OUTPUT
```

not hundred percent of the `$(executable)` syntax works or if we need to put `make release-image-name` within backticks `, or do something like `make release-image-name > $IMAGE_NAME`

Reply:
Gotcha, adding to next commit